packages/block/src/header.ts

```diff
@@ -442,21 +458,32 @@ export class BlockHeader {
    * @param parentBlockHeader - the header from the parent `Block` of this header
    */
   validateGasLimit(parentBlockHeader: BlockHeader): boolean {
-    const parentGasLimit = parentBlockHeader.gasLimit
-    const gasLimit = this.gasLimit
-    const hardfork = this._getHardfork()
-
-    const a = parentGasLimit.div(
-      new BN(this._common.paramByHardfork('gasConfig', 'gasLimitBoundDivisor', hardfork))
-    )
-    const maxGasLimit = parentGasLimit.add(a)
-    const minGasLimit = parentGasLimit.sub(a)
+    // Pre EIP-1559 gaslimit check
+    if (!this._common.isActivatedEIP(1559)) {
+      const parentGasLimit = parentBlockHeader.gasLimit
+      const gasLimit = this.gasLimit
+      const hardfork = this._getHardfork()
+
+      const a = parentGasLimit.div(
+        new BN(this._common.paramByHardfork('gasConfig', 'gasLimitBoundDivisor', hardfork))
+      )
+      const maxGasLimit = parentGasLimit.add(a)
+      const minGasLimit = parentGasLimit.sub(a)
 
-    return (
-      gasLimit.lt(maxGasLimit) &&
-      gasLimit.gt(minGasLimit) &&
-      gasLimit.gte(this._common.paramByHardfork('gasConfig', 'minGasLimit', hardfork))
-    )
+      return (
+        gasLimit.lt(maxGasLimit) &&
+        gasLimit.gt(minGasLimit) &&
+        gasLimit.gte(this._common.paramByHardfork('gasConfig', 'minGasLimit', hardfork))
+      )
+      // EIP-1559 gasTarget check
+    } else {
+      const parentGasTarget = parentBlockHeader.gasLimit
+      // check if the block changed the gas target too much
+      return (
+        this.gasLimit.lte(parentGasTarget.add(parentGasTarget.divn(1024))) &&
+        this.gasLimit.gte(parentGasTarget.sub(parentGasTarget.divn(1024)))
+      )
```

Comment:
Currently in a discussion but we can probably simplify this a lot; the logic is actually the same as the pre-1559 logic except that the "effective gas limit" is now the gasTarget. (So gas limit divided by 2)

Reply:
Hmm, I would assume that this check should still be in place also with the spec adoption but then with `const parentGasTarget = parentBlockHeader.gasLimit / 2` (what would be the rounding here?)? 🤔